src/libs/ajax/GoogleStorage.ts

```diff
@@ -89,17 +89,17 @@ const fetchBuckets = _.flow(
   withUrlPrefix('https://storage.googleapis.com/')
 )(fetchOk);
 
-/**
- * Only use this if the user has write access to the workspace to avoid proliferation of service accounts in projects containing public workspaces.
- * If we want to fetch a SA token for read access, we must use a "default" SA instead (api/google/user/petServiceAccount/token).
- */
 const getServiceAccountToken: (googleProject: string, token: string) => Promise<string> = Utils.memoizeAsync(
```

Comment:
not part of your PR, but now that I'm looking at this function: the memoization will cache the result for 30 minutes. Is that safe?

Reply:
I think memoizing for 30m is ok. It is keyed by the args which includes the google project. This might be noticeable if a user's permissions change.